### ISSUE TITLE
Tutorial: Chrome extension install via Chrome webstore

### ISSUE THREAD

**Issue by StreamTitan** [jitsi/jidesha#39]
Tutorial for doing an inline install with the Chrome extension via the Chrome webstore.
This is different to the manual install which you users use the crx file to set the extension. This inline install is preferable because you do not need to provide a user with the extension. When they click the "Screen Share" icon in the conference interface, it will automatically install the extension (inline install).

 1. Go to https://github.com/jitsi/jidesha and download the files.
2. Your download will be a zip file, unzip: "jidesha-master"
3. Open the calendar folder and edit the manifest.json file
4. The lines you need to edit include:
"name", "description", "icons (your png icons 16x16, 48x48 and 128,128), "web_accessible_resources" and 

    "externally_connectable": {
          "matches": [
            "*://www.yourdomain.com/*"
          ]
      },

Remember to put www in front of your domain and not using the domain name only. The reason for this is that when we go to the developer dashboard in Chrome, we need to verify the website (that you have access to it). Once your site is verified, the main will be selectable but from a drop down list. The domain reads as www.yourmain.com in the drop down list and your manifest.json must match this as www.yourdomain.com and yourdomain.com are in this case, NOT the same thing.

5. Remember to replace the png files with your png files AND name them as you have them in your mainfest.json file.
6. Now run the make.sh file (just click it)
7. It will create a new folder "target", inside this another folder "content"
8. The content folder now has all your edits and is ready for the to be added to the Chrome Extension via the Chrome Webstore.
9. Zip the "contents" folder.

- Now we set up the Chrome Webstore to distribute our extension

10. Now go to the developer dashboard for Chrome.
https://chrome.google.com/webstore/developer/dashboard

11. Create you developer account. If you don't have a developer account its a one-time fee of $5.00
Your manifest will automatically populate the info into the Extension page eg, "name", "version", "description", "pngs" etc. but you also do edits as well right there in the panel.

12. Lower on the developer dashboard page for your extension, is the "websites" drop down list. This is a list of all the sites you have verified thus far with your Google account. IF you don't see the domain that you want to have your Jitsi-Meet at, on that list then you have to add this site and verify it.

We continue as if you did not have this site added and verified.
13. At the "websites" drop down list there is below this - "add new site". 
You will then be taken to a Google process for you to add a website as verified. This involves you uploading a file to your domain and then clicking the link.

14. Once your website has been verified it will then show on your developer dashboard page in the websites drop down list. IF its not there, just refresh the page. If you see it on the list, select it.

15. Lower down the page is "Inline Install" and a checkbox for it. Select the checkbox.
This is the "allow process" so that when users click the Screen Share icon in Chrome, it will automatically install the extension (after they give it permission to install). 

- Almost done.

16. Follow the addition process for you Chrome extension to the end and you will then see the publishing is in progress. You have to wait a bit for this, don't rush it, its doing its thing.

17. After it shows as "Published" on your dashboard, click the "more info" link attached to your extension on your developer dashboard. You will then have the Item ID for your extension.

Copy this ID. We need this ID so that the Screen Sharing icon in your conference will direct to this extension in the webstore. If you don't have this ID, how can it get the extension?

18. Now on your server files for your Jitsi-Meet go to and edit:
/etc/jitsi/meet/www.mydomain.com-config.js

19. Look for the following on this file:

    // The ID of the jidesha extension for Chrome.
    desktopSharingChromeExtId: 'Item ID',

    // Whether desktop sharing should be disabled on Chrome.
    desktopSharingChromeDisabled: false,

Enter you Item ID into - desktopSharingChromeExtId: 'Item ID',
BE SURE TO ENTER THE ID INSIDE THE ''

These are the only 2 lines in this file which need to be edited. Save the file and then go to your Jitsi-Meet site. IF you have followed the instructions above to the letter, you will have zero issues with this install and your Chrome browser users will be able to install the extension without doing it manually (drag and drop) or having to go to the webstore.

**Comment by damencho** [comment]
> Remember to put www in front of your domain and not using the domain name only.
I do not agree with this part, as the www is specific to your case. Users are deploying using the main domain, and for example, when using a webserver you can handle the www redirect to the main domain(for example meet.jit.si works, but www.meet.jit.si is not a correct url at all). It should be stated something like: you need to put your exact deployment address (domain or www.domain, exactly how users will be accessing it).

And by the way, the instructions that you are writing belong to the wiki part of the projects, can you try moving them there?

Thanks for putting your experience as instructions, as it will help other users!

**Comment by StreamTitan** [comment]
Actually that may be causing me an issue. As now its crashes when someone joins the room.

I'll edit the tutorial to reflect your advice in the www

**Comment by StreamTitan** [comment]
Haven't been able to find a fix. 

When someone comes into the room (moderator and invitee) the room crashes and reloads.
One person in the room and the room loads fine.

**Comment by StreamTitan** [comment]
From the console I'm getting this as an error when a second person joins the room:
Logger.js:125 [conference.js] <e.value>:  CONFERENCE FAILED: conference.videobridgeNotAvailable

Both of our windows reset. Only if 1 is no longer trying to access the room, does the room load.

**Comment by damencho** [comment]
so this means there is no videobridge running. Try restarting jvb. Check it logs why it is like that or jicofo logs it prints message when removing jvb from the pool of known jvb instances.

**Comment by StreamTitan** [comment]
restarted with:

sudo service jitsi-videobridge restart
sudo service jicofo restart

what's the command to check the jitsi-videobridge or jicofo logs?

**Comment by damencho** [comment]
Cat /var/log/jitsi/jvb.log :)

**Comment by StreamTitan** [comment]
Thanks. That jvb log is 234171 lines long.

I'll give this a test tomorrow from 2 different machines and report how it went.

**Comment by otonxbejo** [comment]
Thank you, this tutorial is working great.
